src/Tmds.Systemd.Logging/JournalLogger.cs

```diff
@@ -141,6 +141,7 @@ private static void AppendState(string fieldName, object state, JournalMessage m
                     var pair = keyValuePairs[i];
                     if (pair.Key == OriginalFormat)
                     {
+                        message.Append(fieldName, state.ToString());
```

Comment:
It's been a while since I wrote this, so I may not remember it all correctly.

I think for formatted log messages we still want to ignore `OriginalFormat`.
And for formatted scope messages, we want those to get formatted using the `Log` `formatter` argument.

Something like:

```cs
if (pair.Key == OriginfalFormat)
{
    if (isScope)
    {
       message.Append(field, formatter(state, null));
    }
    continue;
}
```

Does that make sense?

Reply:
The original format itself is ignored, the "continue" is still there, so this is the place where the formatted message is written -- when it encounters the `OriginalFormat`, it just formats it and writes as with the associated field name ("SCOPE" in this case).
Also, I considered using the formatter passed to the `Log` method, but it depends on the `TState` generic type, which does not align with the scope state type, so it cannot be used to format arbitrary scope data.
I was "inspired" by [this example](https://github.com/dotnet/runtime/blob/f3f23b4bde8b0a449d57ef600d99560ac97a784e/src/libraries/Microsoft.Extensions.Logging.Console/src/SystemdConsoleFormatter.cs#L119).

Reply:
> Also, I considered using the formatter passed to the Log method, but it depends on the TState generic type, which does not align with the scope state type, so it cannot be used to format arbitrary scope data.
I was "inspired" by this example.

Ah yes. I think the scope ends up getting formatted here: https://github.com/dotnet/runtime/blob/912e88d02e7d9beab0f165233e881e7880310912/src/libraries/Microsoft.Extensions.Logging.Abstractions/src/FormattedLogValues.cs#L104.

Shouldn't we skip appending the field when `AppendState` is called from here?

https://github.com/tmds/Tmds.Systemd/blob/e47335d73eab8238fd50b93b5013b1988096f662/src/Tmds.Systemd.Logging/JournalLogger.cs#L123-L126

I think we've already appended the formatted message here:

https://github.com/tmds/Tmds.Systemd/blob/e47335d73eab8238fd50b93b5013b1988096f662/src/Tmds.Systemd.Logging/JournalLogger.cs#L114-L117

Reply:
Oh, that's my bad. I wasn't paying enough attention and skipped over the fact that it will doubly-format the message itself since AppendState is called for "STATE" up there.